src/combinators.ts

```diff
@@ -270,12 +281,13 @@ function catchFailure<
   (
     ...args: Parameters<Fn>
   ) => Awaited<ReturnType<C>> extends never[]
-    ? UnpackData<Fn> extends any[] ? UnpackData<Fn>
-    : Awaited<ReturnType<C>> | UnpackData<Fn>
+    ? UnpackData<Fn> extends any[]
```

Comment:
this looks like prettier, lets keep deno fmt?

Reply:
Yeah, this is far from done. I'm gonna clean it up once it starts to get serious ;)